### ISSUE TITLE
Cannot build in versions after next@14.1.0

### ISSUE THREAD

**Issue by ravinggenius** [vercel/next.js#65429]
### Link to the code that reproduces this issue

https://github.com/ravinggenius/minecraft-tools/tree/updates

### To Reproduce

1. Clone project and checkout the `updates` branch
2. Configure environment: `cp .env .env.local` and fill in the missing values
3. Install exact dependencies: `npm ci`
4. Build project: `npm run build` runs as expected
5. **Install any newer version of Next past `14.1.0`**: `npm install next`
6. Build project again and notice the absolute dump of minified JavaScript in the output

### Current vs. Expected behavior

I expect to be able to install newer version of Next to keep my dependencies up-to-date. However every newer version of Next beyond `14.1.0` terminates the build process early and dumps a ton (2.1MB!) of minified JavaScript into the terminal. I've tried multiple versions of Next `14.1.x` and `14.2.x` with the same result.

I am also unable to run the dev server in any version past `14.1.0`. Doing so produces copious erroneous output in the terminal when requesting a page.

### Provide environment information

```bash
Operating System:
  Platform: linux (ChromeOS)
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Sat, 17 Feb 2024 21:12:19 +0000
Binaries:
  Node: 18.18.2
  npm: 9.8.1
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.2.2
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Developer Experience

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local)

### Additional context

I tested in the latest canary release (`14.3.0-canary.44`) also with the same result. You can find the [full build output in this gist](https://gist.github.com/ravinggenius/7b4a7db318b0f8888dc2954303681a34).

**Comment by ravinggenius** [comment]
I found a work-around, but I still don't understand the root cause. This function (https://github.com/ravinggenius/minecraft-tools/blob/a6ee5f4dba06f870bff9d861df216986cb36332e/src/i18n/server.ts#L41-L61) returns `SupportedLocale` (a string literal union) when run with Next 14.1.0. However in Next 14.1.1 and later, this function inexplicably returns `Promise< SupportedLocale>`. I'd really like to understand why it returns a promise, because nothing in the function appears to use promises. To be clear this function returns a promise in Next 14.1.1 and later, even without being maked `async`.

The work-around is simple: just call the function with `await`. To help remind me that it must be called with `await`, I also added `async`, though this has no effect on what it does.
